### ISSUE TITLE
enchancement: Limit outgoing connections to avoid killing routers

### ISSUE THREAD

**Issue by mateon1** [ipfs/go-ipfs#3311]
#### Description:

As of now, IPFS tries to connect to as many peers as it can.
Unfortunately, most of the time it can connect to too many peers.

Right now, I have 3 daemons on my network, all of which have >170 connections.
This, for some reason kills/blocks/congests routers.

Right now, I can only rarely create new outgoing connections, which breaks a lot of services, causes DNS requests to fail 50% (or more) of the time, and many more issues.

This is not an issue of bandwidth, but adding a limit on the amount of connections will help to limit bandwidth as well.

**Comment by mateon1** [comment]
Clarification: This should be a config option, not a hard limit on the user.

**Comment by lidel** [comment]
In 2017 [connection manager](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#connmgr) shipped in [v0.4.12](https://github.com/ipfs/go-ipfs/blob/master/CHANGELOG.md#0412-2017-11-09).

The current default (600-900 connections) may be too high, but can be set to lower values in config file:

```console
$ ipfs config --json Swarm.ConnMgr '{"HighWater": 100, "LowWater": 32, "Type": "basic", "GracePeriod": "15s"}'
```
Values above are just a guess: would be useful to get some real feedback from people with old routers and provide safe values under a predefined profile.

**Comment by Stebalien** [comment]
Closing this in favor of the "don't kill routers" issue (#3320).
